app/controllers/users/products_controller.rb

```diff
@@ -1,11 +1,27 @@
 class Users::ProductsController < ApplicationController
+  before_action :load_category, only: :index
+
   def index
-    @products = Product.by_price(params[:prices])
+    @products = Product
+      .by_category(@ids)
+      .by_price(params[:prices])
       .by_brand(params[:brands])
       .by_color(params[:colors])
       .page(params[:page])
       .per(Settings.record_per_page)
   end
 
   def new; end
+
+  def load_category
+    category = Category.find_by(id: params[:category_id])
+    if category
+      @id = []
+      @ids = category.load_ids(category, @id)
```

Comment:
```suggestion
      @ids = category.load_ids(category, []).flatten luôn e
```
làm gì cần 3 dòng code

Reply:
vâng ạ